docs/interpreter_checklist.md

```diff
@@ -0,0 +1,63 @@
+# StableHLO Interpreter Checklist
+
+In this document, we summarize the guidelines while implementing and reviewing
+interpreter for an op. We have intentionally tied a few auxiliary action items,
+related to verifier and type inference, with the idea of making progress in
+those fronts alongside the interpreter implementation.
+
+1. While implementing the interpreter for an op, double check with related
+   implementations (e.g.
+   [hlo_interpreter](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/hlo/evaluator)
+   and
+   [mlir_interpreter](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/compiler/xla/mlir_hlo/tools/mlir_interpreter)).
+   This action item will allow mutual improvements.
+1. Replace the comments, related to input/output constraints, in
+   the implementations, specifically for verifiers and type inference, with
+   some unique identifier containing "constraint-labels", e.g. `Cn` in
+   [here](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#constraints-2)
+   , used in the
+   [specification](https://github.com/openxla/stablehlo/blob/main/docs/spec.md).
+   The unique identifier should have the format `<op_name>_Cn`.
+   This is to explicitly align all the implementations with the specification:
+   the single source of truth, and help to track gaps in those implementations
+   back to the spec.
+
+   **Why Unique**: The fact that the spec will be augmented with Dynamism,
+   Quantization can affect the suffix number in the constraint label or might
+   change it to something else. A unique identifier will enable faster updates
+   to comments we are planning to add here.
+   1. It is OK to have a comment with multiple constraint-labels or to have
+      multiple comments with the same constraint-label. It all depends on how
+      the constraint-label(s) are coded in the implementations.
+   1. In case there is a mismatch between the constraints in the
+      implementation VS those in the specification, make sure there is an
+      open issue reflecting that discrepancy.
+1. Make sure that the there are sufficient tests to cover for the
+   implementations (e.g. verifiers, type inference, interpreter) of the
+   specification. Specifically,
+   1. For interpreter: Write tests following the
```

Comment:
Can you also add a section to:
1. Make sure the spec only has 1 example
2. The spec example is the same as what is in the ODS (it may not be the same if the spec has > 1 examples)
3. Add a link for "More Examples" in the spec to the interpreter tests
4. Update the interpreter column in status.md to `yes`
5. Default the spec test to be the first test in the `interpreter_foo.mlir`
6. Make sure to use the actual name of the inputs/outputs instead of defaulting to SSA values (e.g. %0, %1, etc.)

Reply:
We could also add another action item to standardize summary in the ODS to address #611

Reply:
Thanks!
I have included all the comments except for
1 and 2: IIRC as part of spec work we already did that 
5. The tests in the spec are not written following the testing guidelines. I am afraid that they can deviate. Plus, IMO with "More Examples" we are expected to add new information.  

Please let me know your thoughts

Reply:
Actually, the spec work items were introduced after we started the spec work, so we still may have (I haven't gone through all the ops) ops with more than one example (which is the case in https://github.com/openxla/stablehlo/pull/996/files). By removing the example which is also in ODS, they also have to be changed.

Yes, I also realized this, but I also think it is good to include it to:
1. Make sure the spec example test we wrote is working (aka no typos in the spec example)
2. Since the spec example already covers most, if not all, the interpreter nuances (for example adjusted start index for DynamicUpdateSliceOp), creating another similar test and excluding the spec example in the test is a bit odd imo. And I agree, we should still be adding new tests.

The decision is still up for discussion though. Lmk your thoughts.